### PR TITLE
Update Portal API spec

### PR DESCRIPTION
**Pull request by edsonmichaque**
### For internal users - Please add a Jira DX PR ticket to the subject!
<br>

---

### Preview Link

<br>

### Description

<br>

#### Screenshots (if appropriate)
<br>

## Checklist

- [ ] I have added a **preview link** to the PR description.
- [ ] I have reviewed the suggestions made by our AI (PR Agent) and updated them accordingly (spelling errors, rephrasing, etc.)
- [ ] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [ ] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [ ] Make sure you have started *your change* off *our latest `master`*.
- [ ] I **labeled** the PR

___

### **PR Type**
documentation

___

### **Description**
- Updated the OpenAPI specification for the enterprise developer portal, enhancing the documentation for API endpoints.
- Added new endpoints and adjusted existing ones to reflect the latest API changes.
- Improved the documentation for API parameters and responses to provide clearer guidance for developers.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>enterprise-developer-portal-swagger.yaml</strong><dd><code>Update and enhance Portal API OpenAPI specification</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/assets/others/enterprise-developer-portal-swagger.yaml

<li>Updated the OpenAPI specification for the portal API.<br> <li> Added new endpoints and adjusted existing ones.<br> <li> Improved documentation for API parameters and responses.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5399/files#diff-015c718543dbd3e884dc650ed9bec9be3628cf8db439061e81b4ebf914a68295">+2976/-1099</a></td>

</tr>                    
</table></td></tr><tr><td><strong>Additional files (token-limit)</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>enterprise-developer-portal-swagger.yaml</strong><dd><code>...</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/assets/others/enterprise-developer-portal-swagger.yaml

...

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5399/files#diff-015c718543dbd3e884dc650ed9bec9be3628cf8db439061e81b4ebf914a68295">+2976/-1099</a></td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by dcs3spp** [comment]
Thanks for submitting PR with updated Portal API spec @edsonmichaque 

There a large number of changes in the PR included, maybe as a result of linter etc.

- The API spec does not render correctly in https://editor.swagger.io/ 
- When the PR is ready for review can you indicate which new/updated endpoints are required for review

File: tyk-docs/assets/others/enterprise-developer-portal-swagger.yaml [hunk 1]
```diff
@@ -2203,62 +2281,283 @@ paths:
...
+                $ref: "#/components/schemas/ProductClientTypeResponse"
           description: OK
         "404":
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/Status'
-          description: Client type is not found
+                $ref: "#/components/schemas/Status"
+          description: Client type is wot found
```
Comment (letzya) [inline tyk-docs/assets/others/enterprise-developer-portal-swagger.yaml:2291]: typo. Also the description could be improved

**Comment by letzya** [comment]
@edsonmichaque what is the plan for this PR? why is it still in draft

**Comment by edsonmichaque** [comment]
> @edsonmichaque what is the plan for this PR? why is it still in draft

@letzya let me close this PR, we don't need it any more

**Pull request closed without merging**